src/test/java/org/codeforamerica/shiba/output/applicationinputsmappers/ParentNotLivingAtHomeInputsMapperTest.java

```diff
@@ -0,0 +1,90 @@
+package org.codeforamerica.shiba.output.applicationinputsmappers;
+
+import org.codeforamerica.shiba.application.Application;
+import org.codeforamerica.shiba.output.ApplicationInput;
+import org.codeforamerica.shiba.output.ApplicationInputType;
+import org.codeforamerica.shiba.pages.data.ApplicationData;
+import org.codeforamerica.shiba.pages.data.InputData;
+import org.codeforamerica.shiba.pages.data.PageData;
+import org.codeforamerica.shiba.pages.data.PagesData;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ParentNotLivingAtHomeInputsMapperTest {
+    ParentNotLivingAtHomeInputsMapper mapper = new ParentNotLivingAtHomeInputsMapper();
+
+    @Test
+    void shouldCreateListOfParentNamesForCorrespondingChildren() {
+        ApplicationData applicationData = new ApplicationData();
+        PagesData pagesData = new PagesData();
+        PageData childrenInNeedOfCarePage = new PageData();
+        childrenInNeedOfCarePage.put("whoNeedsChildCare", InputData.builder()
+                .value(List.of("childAFirstName childALastName 939dc33-d13a-4cf0-9093-309293k3", "childBFirstName childBLastName b99f3f7e-d13a-4cf0-9093-23ccdba2a64d", "childCFirstName childCLastName b99f3f7e-d13a-4cf0-9093-4092384"))
+                .build());
+        PageData parentsNotLivingAtHome = new PageData();
+        parentsNotLivingAtHome.put("whatAreTheParentsNames", InputData.builder()
+                .value(List.of("parentAName", "parentCName"))
+                .build());
+        parentsNotLivingAtHome.put("childIdMap", InputData.builder()
+                .value(List.of("939dc33-d13a-4cf0-9093-309293k3", "b99f3f7e-d13a-4cf0-9093-4092384"))
+                .build());
+
+        pagesData.put("childrenInNeedOfCare", childrenInNeedOfCarePage);
+        pagesData.put("parentNotAtHomeNames", parentsNotLivingAtHome);
+        applicationData.setPagesData(pagesData);
```

Comment:
For the future you can use `PagesDataBuilder`/`PageDataBuilder` to construct `PagesData` rather than building it from scratch. For example, you could declare what you had here in a (hopefully) easier to read/write way:
```java
PagesData pagesData = new PagesDataBuilder().build(List.of(
                new PageDataBuilder("childrenInNeedOfCare", Map.of(
                        "whoNeedsChildCare", List.of("child a", "child b"),
                )),
                new PageDataBuilder("parentNotAtHomeNames", Map.of(
                        "whatAreTheParentsNames", List.of("parent a", "parent b")
                        "childInMap", List.of("child a", "child b")
                ))
        ));
```